internal/pkg/service/common/etcdop/op/txn.go

```diff
@@ -368,6 +368,33 @@ func (v *lowLevelTxn[R]) addElse(op etcd.Op, mapper MapFn) {
 }
 
 func (v *lowLevelTxn[R]) mergeTxn(ctx context.Context, op Op) error {
+	// Step down to a nested Merge operation, if there is no processor/else branch.
+	if txn, ok := op.(txnInterface); ok && isSimpleTxn(txn) {
+		thenStart := len(v.thenOps)
+		elseStart := len(v.elseOps)
+		if err := v.addParts(ctx, txn); err != nil {
+			return err
+		}
+
+		if txn.txnProcessorsCount() > 0 {
+			thenEnd := len(v.thenOps)
+			elseEnd := len(v.elseOps)
+			v.processors = append(v.processors, func(ctx context.Context, r *TxnResult[R]) {
+				response := &etcd.TxnResponse{Succeeded: r.Succeeded()}
+				if r.Succeeded() {
+					response.Responses = r.Response().Txn().Responses[thenStart:thenEnd]
```

Comment:
It is in code below, in responses.
In  code below it is set up like that
```
thenStart := len(v.thenOps)
thenEnd := thenStart + len(thenOps)
```
however this slincing looks weird. `thenStart:thenEnd` => `len(v.thenOps):len(v.thenOps)`.  Looks like we evaluate the transaction when succeeded and present only the last response?

Isn't it possible to get index out of range panic?

Reply:
- Slice ranges are correct, if the transaction was successful, then there are indexes with results for THEN operations, if not, then for ELSE operations.
- Panic cannot occur there, processors are executed only when there is no etcd error, that is, the operation has been executed and its result is being processed.
   - **To make it more clear I added commits:**
      - [tests: Add TestTxnOp_Merge_Nested_ServerError](https://github.com/keboola/keboola-as-code/pull/1788/commits/7a30ab2a83a4edfff834108459b52b302ffe954f)
      - [docs: Info about processors and logical errors](https://github.com/keboola/keboola-as-code/pull/1788/commits/02d3bde2cd901e0c375008edbc41f8bdfa3395ee)
      
-------------------------------      
      
The main issue here (which you did not point out) is that the `response` variable is no longer used anywhere.
- **So it can be simplified** 
   - [fix: Simplify mergeTxn](https://github.com/keboola/keboola-as-code/pull/1788/commits/320d5cebbb036082afebbcd6994d3f0e54aeadef)
- The code was created before the [refactor: Remove unused Txn aggregated sub-results](https://github.com/keboola/keboola-as-code/pull/1788/commits/39c9e7228f91b983abffa629854e726034864095)
   - And I haven't updated it.
   - The response was originally needed for the sub-results.